extensions/2.0/Khronos/KHR_materials_clearcoat/README.md

```diff
@@ -0,0 +1,96 @@
+
+## Extending Materials
+
+The PBR clearcoat materials are defined by adding the `KHR_materials_clearcoat` extension to any compatible glTF material (excluding those listed above).  For example, the following defines a material like varnish using clearcoat parameters.
+
+```json
+{
+    "materials": [
+        {
+            "name": "varnish",
+            "extensions": {
+                "KHR_materials_clearcoat": {
+                    "clearcoatFactor": 1.0
+                }
+            }
+        }
+    ]
+}
+```
+
+### Clearcoat
+
+All implementations should use the same calculations for the BRDF inputs. Implementations of the BRDF itself can vary based on device performance and resource constraints. See [Appendix B](/specification/2.0/README.md#appendix-b-brdf-implementation) for more details on the BRDF calculations.
+
+|                                  | Type                                                                            | Description                            | Required             |
+|----------------------------------|---------------------------------------------------------------------------------|----------------------------------------|----------------------|
+|**clearcoatFactor**               | `number`                                                                        | The clearcoat layer intensity.         | No, default: `0.0`   |
+|**clearcoatTexture**              | [`textureInfo`](/specification/2.0/README.md#reference-textureInfo)             | The clearcoat layer intensity texture. | No                   |
+|**clearcoatRoughnessFactor**      | `number`                                                                        | The clearcoat layer roughness.         | No, default: `0.0`   |
+|**clearcoatRoughnessTexture**     | [`textureInfo`](/specification/2.0/README.md#reference-textureInfo)             | The clearcoat layer roughness texture. | No                   |
+|**clearcoatNormalTexture**        | [`normalTextureInfo`](/specification/2.0/README.md#reference-normaltextureinfo) | The clearcoat normal map texture.      | No                   |
+  
+The clearcoat formula `f_clearcoat` is computed using the specular term from the glTF 2.0 Metallic-Roughness material, defined in [Appendix B](/specification/2.0/README.md#appendix-b-brdf-implementation).  Use F0 equal `0.04`, base color white `1.0, 1.0, 1.0`, metallic value `0.0`, and the clearcoat roughness value defined in this extension as follows:
+
+```
+clearcoatRoughness = clearcoatRoughnessFactor * clearcoatRoughnessTexture.g
+```
+
+The following abstract code describes how the base and clearcoat layers should be blended together:
+  
+```
+clearcoatWeight = clearcoatTexture.r * clearcoatFactor * fresnel(0.04, NdotV)
+
+f = mix(f_emissive + f_diffuse + f_specular, f_clearcoat, clearcoatWeight)
+```
+
+If `clearcoatFactor = 0`, the clearcoat layer is disabled.
```

Comment:
Minor correction: replace `clearcoatFactor` here with `clearcoatWeight`.

Reply:
`clearcoatWeight` can differ per-pixel.  But if `clearcoatFactor` is zero in the JSON, then the entire extension can be considered disabled.